snappy/snap_file.go

```diff
@@ -229,6 +229,14 @@ func (s *SnapFile) Install(inter progress.Meter, flags InstallFlags) (name strin
 	if err := s.m.addSquashfsMount(s.instdir, inhibitHooks, inter); err != nil {
 		return "", err
 	}
+	// if anything goes wrong we ensure we stop
+	defer func() {
+		if err != nil {
+			if e := s.m.removeSquashfsMount(s.instdir, inter); e != nil {
```

Comment:
A personal question about go style here. So far, I found that the common thing to do is  to overwrite the err variable. Is there a reason here to declare a new var e?